Source/WebCore/page/FrameView.cpp

```diff
@@ -3572,6 +3575,8 @@ void FrameView::performPostLayoutTasks()
     }
 
     scrollToAnchor();
+    if (m_scrollAnchoringController)
+        m_scrollAnchoringController->updateScrollPosition();
```

Comment:
I know scrollToAnchor() is here, but this can run multiple times between rendering updates because of forced layouts. A better time to do this is inside `document.runScrollSteps();` (and I'd expect the spec to say that?).

Reply:
I tried adding it to document.runScrollSteps() (I think you are right that it should go here), but it was causing the tests to fail even though I was making the call to `m_frameView.setScrollPosition()` with the correct offset. My assumption is that we need to trigger a render update after the scroll right? Calling `page()->scheduleRenderingUpdate()` didn't seem to affect this.

Reply:
Did the tests fail because they were written in a way that expects some kind of synchronous anchoring behavior?

I would expect that the triggering of scroll events already schedules a rendering update after a scroll.

Reply:
I added a call to updateScrollPositionForAnchorElement in Document::updateLayoutIgnorePendingStylesheets to ensure calls to scrollTop/etc would get the current state including any adjustments due to scroll anchoring.